reference.md

```diff
@@ -499,18 +542,154 @@ variables are missing:
 
 ``` fennel
 ;; bad
-(match [1 2 3]
+(case [1 2 3]
   ;; Will throw an error because `b' is nil for the first
   ;; pattern but the guard still uses it.
   (where (or [a 1 2] [a b 3]) (< a 0) (< b 1))
   :body)
 
 ;; ok
-(match [1 2 3]
+(case [1 2 3]
   (where (or [a b 2] [a b 3]) (< a 0) (<= b 1))
   :body)
 ```
 
+#### Binding Unification
+
+Symbols bound inside a `case` pattern are independent from any existing
+symbols in the current scope, that is - names may be re-used without
+consequence.
+
+Example:
+
+```fennel
+(let [x 1]
+  (case [:hello]
+    ;; `x` is simply bound to the first value of [:hello]
+    [x] x)) ; -> :hello
+```
+
+Sometimes it may be desirable to match against an existing value in the outer
+scope. To do this we can "unify" a binding inside the pattern with an existing
+outer binding with the unary `(= my-symbol)` form. The unary `(= my-symbol)`
+form is *only* valid in a `case` pattern.
+
+Example:
+
+```fennel
+(let [x 1]
+  (case [:hello]
+    ;; 1 != :hello
+    [(= x)] x
+    _ :no-match)) ; -> no-match
+
+(let [x 1]
+  (case [1]
+    ;; 1 == 1
+    [(= x)] x
+    _ :no-match)) ; -> 1
+
+(let [pass :hunter2]
+  (case (user-input)
+    (= pass) :login
+    _ :try-again!))
+```
+
+Unification is only required inside the pattern. Outer bindings are
+automatically available inside guards and bodies as long as the name has not
+been rebound in the pattern.
```

Comment:
I'm not sure what "Unification is only required inside the pattern" here means.

I also think we may need to re-evaluate the way we use the term "unification" as it has both a more narrow meaning than how we use it (https://lists.sr.ht/~technomancy/fennel/%3C3j4SZ0Dbj8uiJhME0MPDVUxqBNlNEg1IY9Lv3LPWP_ZZb7EyEgSMvxRXb_TF_TzpT2zGSKs-cIoMuYYMFYrzCcfBE6YAD36i-rvSUuWSU0I%3D%40protonmail.com%3E) and in other cases a broader meaning: https://blog.fogus.me/2010/12/14/unification-versus-pattern-matching-to-the-death/

Perhaps we need to be very specific that we mean "unification with outer scope"; that might be enough.